app/models/krikri/provider.rb

```diff
@@ -1,80 +1,173 @@
+    #
+    # @param [String] :id or :rdf_subject
     #
     # @return [Krikri::Provider]
-    # @todo Use {ActiveTriples}/{RDF::Repository} to populate the object
     def self.find(id)
-      return nil if id.nil?
-      provider = new(id)
-      provider.reload
-      return nil if provider.empty?
-      provider
+      rdf_subject = id.start_with?(base_uri) ? id : base_uri + id
+      query_params = { :rows => 1,
+                       :q => rdf_subject,
+                       :qf => 'provider_id',
+                       :fl => 'provider_id provider_name' }
+      response = Krikri::SolrResponseBuilder.new(query_params).response
+      return nil unless response.docs.count > 0
+      provider = new({ :rdf_subject => rdf_subject,
+                       :name => response.docs.first['provider_name'].first })
     end
 
     ##
-    # Loads the provider's data from the repository.
-    #
-    # @return [Krikri::Provider] self
-    def reload
-      query = Krikri::Repository.query_client.select.distinct.where([self, :p, :o])
-      query.each_solution do |solution|
-        set_value(solution.p, solution.o)
-      end
+    # Get the base of the :rdf_subject for any provider
+    # @return [String] ending in "/"
+    def self.base_uri
+      base_uri = Krikri::Settings.prov.provider_base
+      base_uri.end_with?("/") ? base_uri : base_uri << "/"
+    end
+
+    def id
+      @id ||= local_name
+    end
+
+    def name
+      @name ||= initialize_name
+    end
 
-      self
+    def agent
+      @agent ||= initialize_agent
     end
 
     ##
-    # A list of records of type ore:Aggregation associated with this provider
-    #
-    # @return [Array<DPLA::MAP::Aggregation>] the Aggregations with this
-    #   provider
-    def records
-      query = Krikri::Repository.query_client.select
-              .where([:record, RDF::EDM.provider, self],
-                     [:record, RDF.type, RDF::ORE.Aggregation])
+    # Tests for providers that have valid a :rdf_subject (not a bnode).
+    # A valid :rdf_subject does not necessarily match and :rdf_subject in the
+    # Solr index, but it has the correct URI format.
+    def valid_rdf_subject?
+      return false unless rdf_subject.present?
+      rdf_subject.start_with?(self.class.base_uri) ? true : false
+    end
 
-      query.execute.map do |solution|
-        DPLA::MAP::Aggregation.new(solution.record)
-      end
+    ##
+    # Required ActiveModel method.
+    def persisted?
+      false
     end
 
+    private
+
     ##
-    # @return [String] the local name (last uri segment) for the provider
-    def id
-      node? ? nil : rdf_subject.to_s.gsub(base_uri.to_s, '')
+    # Gives the last path fragment for :rdf_subject in HTML escaped form,
+    # ignoring :rdf_subjects, ignoring bnodes.
+    #
+    # @example
+    #   Given: rdf_subject == 'http://my_domain/blah/0123'
+    #   local_name == '0123'
+    #
+    # @return [String]
+    def local_name
```

Comment:
Yes - I am hoping that this will be fixed when we change the way that `current providers` are tracked in the new IA.